numpy/core/_methods.py

```diff
@@ -230,3 +231,14 @@ def _ptp(a, axis=None, out=None, keepdims=False):
         umr_minimum(a, axis, None, None, keepdims),
         out
     )
+
+def _dump(self, file, protocol=2):
+    if hasattr(file, 'write'):
+        ctx = contextlib_nullcontext(file)
+    else:
+        ctx = open(os_fspath(file), "wb")
+    with ctx as f:
+        pickle.dump(self, file, protocol=protocol)
+
+def _dumps(self, protocol=2):
+    pickle.dumps(self, protocol=protocol)
```

Comment:
missing return in both these functions, which would seem to indicate missing or incomplete tests for `arr.dumps()` and `arr.dump()`

Reply:
@mattip doesn't that just return None?

Reply:
yes, but `dumps` is supposed to return the string

Reply:
Wow, that was sloppy. You're right, dumps clearly has no tests.

`pickle.dump` has no return value, so that one doesn't really matter.